musl-compatibility.h

```diff
@@ -11,4 +11,14 @@
     _rc; })
 # endif
 
+/* GNU basename() like functionality as a macro
+ *
+ * Used for gnu vs posix basename() disambiguation.
+ */
+#define simple_basename(PATH) ({\
```

Comment:
How about the idea of just using glib and avoid the whole problem of different behavior on basename()? g_path_get_basename() would require some deallocation of the result but it should be nicely consistent regardless of libc.

Reply:
That would be an option, but it would require "real changes" in code and some standalone things might not use glib and ... vs. just making things musl compatible with minimal diff noise.

Reply:
Ok, well I'll just approve, we've spent enough time here. I'll approve and update my cleanup PR to have only changes not related to basename().